aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/AWSS3StoragePlugin.java

```diff
@@ -190,13 +220,17 @@ public AmazonS3Client getEscapeHatch() {
             @NonNull String local,
             @NonNull StorageUploadFileOptions options,
             @NonNull Consumer<StorageUploadFileResult> onSuccess,
-            @NonNull Consumer<StorageException> onError) {
-
+            @NonNull Consumer<StorageException> onError
+    ) {
         AWSS3StorageUploadFileRequest request = new AWSS3StorageUploadFileRequest(
                 key,
                 local,
-                options.getAccessLevel() != null ? options.getAccessLevel() : defaultAccessLevel,
-                options.getTargetIdentityId(),
+                options.getAccessLevel() != null
+                        ? options.getAccessLevel()
+                        : defaultAccessLevel,
+                options.getTargetIdentityId() != null
+                        ? options.getTargetIdentityId()
+                        : userIdentityId,
```

Comment:
Although this ternary is simple logic, it is repeated enough times that we might as well have a `getTargetIdentityId()` and `getUserIdentityId()`, that we can in-line to all of these request creations.

Reply:
I thought about it, but that required passing in `options` parameter to the helper method also, which kind of looked ugly.